### ISSUE TITLE
Incorrect 'CS8629 Nullable value type may be null' on implicitly-cast operand of lifted operator

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#39361]
**Version Used**: New in VS 16.4p2, not present in 16.4p1 or 16.3.5

No warning should be shown because the operator is lifted and therefore expects nulls:

```cs
#nullable enable

struct Foo
{
    public static implicit operator DateTime?(Foo? value) => default;

    bool M(Foo? x, DateTime y)
    {
        // CS8629 Nullable value type may be null.
        //     ↓
        return x == y;
    }
}
```

The other operators also trigger this bug.

**Comment by jnm2** [comment]
This is still a problem in 16.4-p4.

**Comment by jnm2** [comment]
This has now become a regression from 16.3 to 16.4 that broke our build.
